File: eat-it/src/main/java/ide/eatit/model/responses/ItemResponse.java
package ide.eatit.model.responses;

public class ItemResponse<T> extends BaseResponse {

    private T item;

    public ItemResponse(String statusCode, String message, String status, T item) {
        super(statusCode, message, status);
        this.item = item;
    }

    public T getItem() {
        return item;
    }

    public void setItem(T item) {
        this.item = item;
    }
}

File: eat-it-fe/src/app/models/responses/item-response.ts
import { BaseResponse } from "./base-response";

export class ItemResponse<T> extends BaseResponse {
  item: T;

  constructor(statusCode: string, message: string, status: string, item: T) {
    super(statusCode, message, status);
    this.item = item;
  }

  static override failedResponse<Z>(message: string = "", statusCode: string = '500'): ItemResponse<Z> {
    return new ItemResponse<Z>(statusCode, "Failed to ItemResponse. " + message, 'FAILED', Object.create(null));
  }
}